torchvision/datasets/folder.py

```diff
@@ -148,23 +202,9 @@ def make_dataset(
     ) -> List[Tuple[str, int]]:
         return make_dataset(directory, class_to_idx, extensions=extensions, is_valid_file=is_valid_file)
 
-    def _find_classes(self, dir: str) -> Tuple[List[str], Dict[str, int]]:
-        """
-        Finds the class folders in a dataset.
-
-        Args:
-            dir (string): Root directory path.
-
-        Returns:
-            tuple: (classes, class_to_idx) where classes are relative to (dir), and class_to_idx is a dictionary.
-
-        Ensures:
-            No class is a subdirectory of another.
-        """
-        classes = [d.name for d in os.scandir(dir) if d.is_dir()]
-        classes.sort()
-        class_to_idx = {cls_name: i for i, cls_name in enumerate(classes)}
-        return classes, class_to_idx
+    @staticmethod
+    def _find_classes(dir: str) -> Tuple[List[str], Dict[str, int]]:
```

Comment:
nit: I wouldn't made this a `staticmethod`, as other instantiations of this dataset could rely on `self` for generating a custom set of classes and class ids

But this is not really important, so let's move forward with this PR and get this merged